protocol/x/clob/keeper/process_operations.go

```diff
@@ -675,6 +675,26 @@ func (k Keeper) PersistMatchDeleveragingToState(
 	}
 	deltaBaseQuantumsIsNegative := position.GetIsLong()
 
+	// TODO: This logic assumes that the subaccount is negative equity. Should account for final settlement markets.
+	// If there are zero-deleveraging fills, this is a sentinel value to indicate a subaccount could not be liquidated
+	// or deleveraged and still has negative equity. Mark the current block number in state to indicate a negative TNC
+	// subaccount was seen.
+	if len(matchDeleveraging.GetFills()) == 0 {
```

Comment:
Question: should validation be added that this only happens once per block?

I punted on this to keep things simple and there doesn't seem to be much benefit to adding this validation 🤔